BusBoard.ConsoleApp/Apis/Tfl/TflApi.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using RestSharp;
+
+namespace BusBoard.ConsoleApp
+{
+    class TflApi
+    {
+        private const string BaseUrl = "https://api.tfl.gov.uk/";
+
+        private readonly IRestClient client;
+
+        public TflApi()
+        {
+            client = new RestClient(BaseUrl);
+        }
+
+        public T Execute<T>(RestRequest request) where T : new()
+        {
+            var response = client.Execute<T>(request);
+
+            if (response.ErrorException != null)
+            {
+                throw response.ErrorException;
+            }
+            else if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                throw new TflStopNotFoundException("Stop was not found");
+            }
+            else if (response.StatusCode == HttpStatusCode.BadRequest)
+            {
+                throw new TflStopNotFoundException("Location requested is too far from London");
```

Comment:
Is this the only way the request can be invalid? 🙂

Reply:
I'm fairly sure that's the only way the request can be invalid due to user input (a `BadRequest`), because it's plumbed in from the postcode finder, so the two values going in there definitely are doubles, and are valid coordinates (assuming Postcode.io doesn't give out invalid coordinates). After this processing, the only way the request could be invalid that I can think of is if the postcode finder provides coordinates outside of London, unless I've missed something?

However the API might change or be down, so I should probably add some default error handling for any other non-OK status codes

Reply:
Sure, that sounds sensible 🙂

Reply:
Added default error handling in 0ac0101 🙂